torch_xla/csrc/ops/index_ops.cpp

```diff
@@ -221,12 +234,18 @@ ir::NodePtr IndexCopyOp(const ir::Value& buffer, xla::int64 dim,
 
 }  // namespace
 
-CanonicalIndexInfo GetCanonicalIndexInfo(const at::Tensor& base,
-                                         at::TensorList orig_indices) {
+CanonicalIndexInfo GetCanonicalIndexInfo(
+    const at::Tensor& base,
+    const c10::List<c10::optional<at::Tensor>>& orig_indices) {
   CheckIndexTensorTypes(orig_indices);
   // First expand ByteTensor (boolean masks) into 1 or more LongTensors, then
   // broadcast all index tensors together.
-  auto indices = at::expand_outplace(ExpandByteTensors(base, orig_indices));
+  auto indices_vec = at::expand_outplace(ExpandByteTensors(base, orig_indices));
+  torch::List<c10::optional<at::Tensor>> indices;
+  indices.reserve(indices_vec.size());
+  for (at::Tensor i : std::move(indices_vec)) {
+    indices.push_back(std::move(i));
+  }
```

Comment:
Why couldn't we just pass `indices_vec`? I didn;t find the function prototype change of `expand_outplace ` in the linked pytorch pr nor can I find the `expand_outplace` that takes `c10::List<c10::optional<at::Tensor>>&` in https://github.com/pytorch/pytorch/blob/71ca600af93b4ad58ca6fc083fa5d42a56b43194/aten/src/ATen/ExpandUtils.h. What am I missing here?

Reply:
you're right, I changed it